mne/epochs.py

```diff
@@ -3249,6 +3250,14 @@ def __init__(
         # keep track of original sfreq (needed for annotations)
         raw_sfreq = raw.info["sfreq"]
 
+        # get events from annotations if no events given
+        if events is None:
+            events, event_id_tmp = events_from_annotations(raw)
+            if events.size == 0:
+                raise RuntimeError("No events found in raw.annotations")
```

Comment:
This message is a little bit of a misomer, no?

Also how can I constrain the annotations that will be used to construct epochs?

Or would your idea be to select the desired subset of epochs after creation?

Reply:
> Are you thinking no `onsets` found?

No, more like, "no annotations found" ;)




> I was thinking you could not preload so then it would be no big deal to select after creation but before `load_data`.

Fair enough!

Reply:
I changed this to allow `event_id` to match the `description` for constraining after looking through examples in the documentation, I think this is pretty elegant.